xwiki-rendering-macros/xwiki-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/AbstractMessageMacro.java

```diff
@@ -62,28 +53,38 @@
     public AbstractMessageMacro(String macroName, String macroDescription)
     {
         super(macroName, macroDescription,
-            new DefaultContentDescriptor("Content of the message", true, Block.LIST_BLOCK_TYPE));
+            new DefaultContentDescriptor("Content of the message", true, Block.LIST_BLOCK_TYPE),
+            BoxMacroParameters.class);
     }
 
     @Override
-    public List<Block> execute(Object parameters, String content, MacroTransformationContext context)
-        throws MacroExecutionException
+    public List<Block> execute(BoxMacroParameters parameters, String content,
+        MacroTransformationContext context) throws MacroExecutionException
     {
         if (StringUtils.isEmpty(content)) {
             throw new MacroExecutionException(CONTENT_MISSING_ERROR);
         }
+        return super.execute(parameters, content, context);
+    }
 
-        BoxMacroParameters boxParameters = new BoxMacroParameters();
-
-        boxParameters.setCssClass(context.getCurrentMacroBlock().getId() + "message");
-
-        // TODO: we should not rely directly on the concrete macro, but only on the AbstractBoxMacro
-        return this.boxMacro.execute(boxParameters, content, context);
+    @Override
+    protected List<Block> parseContent(BoxMacroParameters parameters, String content,
+        MacroTransformationContext context) throws MacroExecutionException
+    {
+        List<Block> macroContent = getMacroContentParser().parse(content, context, false, context.isInline())
+            .getChildren();
+        return Collections.singletonList(new MetaDataBlock(macroContent, this.getUnchangedContentMetaData()));
     }
 
     @Override
     public boolean supportsInlineMode()
     {
         return true;
     }
+
+    @Override
+    protected String getClassProperty()
+    {
+        return super.getClassProperty() + ' ' + this.getDescriptor().getId() + "message";
```

Comment:
The macro might be associated to a specific syntax to it's probably safer to do this.getDescriptor().getId().getId().